omnigibson/utils/usd_utils.py

```diff
@@ -283,13 +326,12 @@ def get_all_impulses(cls):
                 in the simulator and M tracked rigid bodies
         """
         # Generate the contact matrix if it doesn't already exist
-        if cls._CONTACT_MATRIX is None:
-            cls._CONTACT_MATRIX = cls._CONTACT_VIEW.get_contact_force_matrix(dt=1.0)
+        if scene_idx not in self._CONTACT_MATRIX:
+            self._CONTACT_MATRIX[scene_idx] = self._CONTACT_VIEW[scene_idx].get_contact_force_matrix(dt=1.0)
 
-        return cls._CONTACT_MATRIX
+        return self._CONTACT_MATRIX[scene_idx]
 
-    @classmethod
-    def get_impulses(cls, prim_paths_a, prim_paths_b):
+    def get_impulses(self, prim_paths_a, prim_paths_b):
```

Comment:
shouldnn't this also take in `scene_id` arg?

Reply:
TODO